src/java/org/apache/cassandra/io/sstable/SSTableZeroCopyWriter.java

```diff
@@ -195,14 +197,16 @@ public void close()
             writer.close();
     }
 
-    public void writeComponent(Component.Type type, DataInputPlus in, long size) throws ClosedChannelException
+    public void writeComponent(Component component, DataInputPlus in, long size) throws ClosedChannelException
     {
-        logger.info("Writing component {} to {} length {}", type, componentWriters.get(type).getPath(), prettyPrintMemory(size));
+        @SuppressWarnings({"resource", "RedundantSuppression"})  // all writers are closed in close()
```

Comment:
Curious...can individual writers fail and cause some writers not to be closed? (above, on 197)

Reply:
I guess if that's the case, we could use `FBUtitilites#closeAll` at `SSTableZeroCopyWriter#close`. But we are not modifying that here, so I think it can be done on a separate ticket.